### ISSUE TITLE
database connection issue upon fresh installation

### ISSUE THREAD

**Issue by milkywade** [wallabag/wallabag#3854]
### Issue details

Following install procedure on documentation.

"make install" fails on wallabag installer, step 1, database connection, with error message "Can't connect to the database: An exception occured in driver: SQLSTATE[HY000] [1045] Access denied for user 'wallabag'@'localhost' (using password: YES)"

### Environment

* wallabag version (or git revision) that exhibits the issue: last as of 14 Jan 2019
* How did you install wallabag? git clone
* Last wallabag version that did not exhibit the issue (if applicable): n/a
* php version:  7.0.33-0+deb9u1
* OS: Linux 5250 4.14.79-v7+ (raspbian)
* type of hosting (shared or dedicated): dedicated
* which storage system you choose at install (SQLite, MySQL/MariaDB or PostgreSQL): MariaDB

### Additional information

this is my parameters.yml file:
# This file is auto-generated during the composer install

parameters:
    database_driver: pdo_mysql
    database_driver_class: null
    database_host: localhost
    database_port: null
    database_name: wallabag
    database_user: wallabag
    database_password: 'thisorthat'
    database_path: null
    database_table_prefix: wallabag_
    database_socket: null
    database_charset: utf8mb4
    domain_name: 'https://something.lan'
    mailer_transport: smtp
    mailer_host: 127.0.0.1
    mailer_user: null
    mailer_password: null
    locale: en
    secret: stuff_here
    twofactor_auth: false
    twofactor_sender: no-reply@localhost
    fosuser_registration: false
    fosuser_confirmation: false
    from_email: no-reply@localhost
    rss_limit: 50
    rabbitmq_host: localhost
    rabbitmq_port: 5672
    rabbitmq_user: guest
    rabbitmq_password: guest
    rabbitmq_prefetch_count: 10
    redis_scheme: tcp
    redis_host: localhost
    redis_port: 6379
    redis_path: null
    redis_password: null

If I connect via command line, it works (I use a 16 char long password):

mysql -u wallabag --password=thisorthat wallabag
Welcome to the MariaDB monitor.  Commands end with ; or \g.
Your MariaDB connection id is 5326
Server version: 10.1.37-MariaDB-0+deb9u1 Raspbian 9.0

Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

MariaDB [wallabag]>

**Comment by techexo** [comment]
I don't have quotes around my password in `parameters.yml`, could it be related? Or is it needed because of special characters, maybe?

**Comment by milkywade** [comment]
thanks, I changed to a short, uncomplicated password and that worked.

HOWEVER: not sure this is a good idea security-wise.

**Comment by techexo** [comment]
@wallabag/core, what is the best way to handle passwords including special characters in parameters.yml? Is there absolutely forbidden chars?

**Comment by j0k3r** [comment]
This is strange. What was the password?

**Comment by milkywade** [comment]
+n9J/vfa@*Xk+\77

**Comment by techexo** [comment]
I didn't make any test to check, but wouldn't the `\` be interpreted as an escaping character somehow?

**Comment by milkywade** [comment]
IMHO The doc may not be enough. The password was provided during make install so if a given char is not supported, the install script should also check syntax for compliance.

cheers

**Comment by techexo** [comment]
I agree with you. I think it's worth trying to figure out where the issue comes from exactly:
- when creating your DB, was your password the one you thought it was. Test for this would be to try and create a database, then check if `# mysql -u wallabag --password=p@ssw()rd wallabag` and `# mysql -u wallabag --password='p@assw()rd' wallabag` allows one to connect ;
- during the install of wallabag, is the password you enter in the script the exact same one than the one stored in `parameters.yml`?
- is there a difference if there are quotes around the password in `parameters.yml`?
- how does PHP handles strings containing special characters? (In python there is "raw strings" that allows for the use of special characters without escaping, maybe that exists in PHP too?)

**Comment by techexo** [comment]
Finally, it is probably an error if passwords are not quoted in `parameters.yml` by default. Some rules of thumbs are available on [SO](https://stackoverflow.com/a/22235064). If we follow these rules, simple quotes would be best, with the exception that when a quote is in the password, it should be doubled.

**Comment by milkywade** [comment]
FYI the password was entered with no quotes during make install which wrote it to parameters.yml with simple quotes.

**Comment by techexo** [comment]
Great. So the issue would have come from either the accidental interpretation of the characters before initiating the connection with the database, or the mysql/mariadb server didn't register the password it actually was supposed to be?
